plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAnalyze.java

```diff
@@ -765,11 +778,232 @@ public void testCollectStatsAfterColumnAdded(boolean collectOnWrite)
+                        ('c_str', 10.0, 4.0, 0.2, null, null, null),
+                        (null, null, null, null, 5.0, null, null)
+                        """);
+
+        // Ensure that ANALYZE does not change data
+        assertQuery(
+                "SELECT * FROM " + tableName,
+                """
+                        VALUES
+                        (42, 'foo'),
+                        (12, 'ab'),
+                        (null, null),
+                        (15, 'cd'),
+                        (15, 'bar')
+                        """);
+
+        deleteRecursively(Path.of(new URI(getTableLocation(tableName)).getPath()), ALLOW_INSECURE);
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testNoColumnStats()
+            throws IOException, URISyntaxException
+    {
+        String tableName = copyResourcesAndRegisterTable("no_column_stats");
+        assertQuery("SELECT * FROM " + tableName, "VALUES (42, 'foo')");
+
+        assertUpdate("ANALYZE " + tableName, 1);
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                """
+                        VALUES
+                        ('c_int', null, 1.0, 0.0, null, 42, 42),
+                        ('c_str', 3.0, 1.0, 0.0, null, null, null),
+                        (null, null, null, null, 1.0, null, null)
+                        """);
+
+        deleteRecursively(Path.of(new URI(getTableLocation(tableName)).getPath()), ALLOW_INSECURE);
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testPartiallyNoStats()
+            throws IOException, URISyntaxException
+    {
+        String tableName = copyResourcesAndRegisterTable("no_stats");
+        // Add additional transaction log entry with statistics
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1,'a'), (12,'b')", 2);
+        assertQuery("SELECT * FROM " + tableName,
+                """
+                        VALUES
+                        (42, 'foo'),
+                        (12, 'ab'),
+                        (null, null),
+                        (15, 'cd'),
+                        (15, 'bar'),
+                        (1, 'a'),
+                        (12, 'b')
+                        """);
+
+        // Simulate initial analysis
+        assertUpdate(format("CALL %s.system.drop_extended_stats('%s', '%s')", DELTA_CATALOG, TPCH_SCHEMA, tableName));
+
+        assertUpdate("ANALYZE " + tableName, 7);
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                """
+                        VALUES
+                        ('c_int', null, 4.0, 0.14285714285714285, null, 1, 42),
+                        ('c_str', 12.0, 6.0, 0.14285714285714285, null, null, null),
+                        (null, null, null, null, 7.0, null, null)
+                        """);
+
+        deleteRecursively(Path.of(new URI(getTableLocation(tableName)).getPath()), ALLOW_INSECURE);
```

Comment:
deleteRecursively should be done after `DROP TABLE` right?

btw all the tests use these two commands, you could create a utility command for this purpose.